takari-lifecycle-plugin/src/main/java/io/takari/maven/plugins/compile/jdt/AnnotationProcessorManager.java

```diff
@@ -125,7 +130,16 @@ public ProcessorInfo discoverNextProcessor() {
     if (processors.hasNext()) {
       Processor processor = processors.next();
       processor.init(_processingEnv);
-      ProcessorInfo procecssorInfo = new ProcessorInfo(processor);
+      ProcessorInfo procecssorInfo = new ProcessorInfo(processor) {
+        @Override
+        public boolean computeSupportedAnnotations(Set<TypeElement> annotations, Set<TypeElement> result) {
+          boolean shouldCall = super.computeSupportedAnnotations(annotations, result);
+          if (shouldCall) {
+            incrementalCompiler.onAnnotationProcessing();
```

Comment:
no, it does not do annotation processing, it tells the caller the annotation processing is necessary. not ideal, but this is close enough and does not require changes to eclipse jdt compiler.